applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py

```diff
@@ -129,11 +130,31 @@ def RunOptimizationLoop(self):
             KM.Logger.PrintInfo("ShapeOpt", "Time needed for current optimization step = ", timer.GetLapTime(), "s")
             KM.Logger.PrintInfo("ShapeOpt", "Time needed for total optimization so far = ", timer.GetTotalTime(), "s")
 
+            if self.__isAlgorithmConverged():
+                break
+
     # --------------------------------------------------------------------------
     def FinalizeOptimizationLoop(self):
         self.analyzer.FinalizeAfterOptimizationLoop()
         self.data_logger.FinalizeDataLogging()
 
+    # --------------------------------------------------------------------------
+    def __isAlgorithmConverged(self):
+
+        if self.opt_iteration > 1 :
+            # Check if maximum iterations were reached
+            if self.opt_iteration == self.algorithm_settings["max_iterations"].GetInt():
+                KM.Logger.Print("")
+                KM.Logger.PrintInfo("ShapeOpt", "Maximal iterations of optimization problem reached!")
+                return True
+
+            # Check for relative tolerance
+            relativeChangeOfObjectiveValue = self.data_logger.GetValues("rel_change_objective")[self.opt_iteration]
```

Comment:
relative_change_of_objective_value - local variables in lower case

Reply:
Ok! I went ahead and updated that in the rest of algorithms for consistency